osbuild/pipeline.py

```diff
@@ -348,15 +343,18 @@ def run(self, store, monitor, libdir, output_directory):
                 if not results["success"]:
                     return results
 
-                r = self.assemble(object_store,
-                                  build_tree,
-                                  tree,
-                                  monitor,
-                                  libdir,
-                                  output_directory)
+                r, obj = self.assemble(object_store,
+                                       build_tree,
+                                       tree,
+                                       monitor,
+                                       libdir)
 
                 results.update(r)  # This will also update 'success'
 
+            if obj and output_directory:
+                obj.export(output_directory)
+                obj.cleanup()
+
```

Comment:
```suggestion
            if obj:
                if output_directory:
                    obj.export(output_directory)
                obj.cleanup()

```

I think you need to call `obj.cleanup()` even if `output_directory` is not set.

Reply:
As a followup, you might be able to drop all the `obj.cleanup()` calls, since the objectstore calls it anyway. Maybe @gicmo can verify that claim ;)

Reply:
True, the `obj.cleanup()` is a more eager cleanup so we free up space earlier. But it is not really needed from a correctness point-of-view.

Reply:
Also if we get rid of the `obj.cleanup()` here, which so so close to the end of the context manager, we can collapse the two `if`s:, i.e.: `if obj and output_directory:`